lib/mongo/error/unsupported_allow_disk_use.rb

```diff
@@ -0,0 +1,38 @@
+# Copyright (C) 2020 MongoDB Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+module Mongo
+  class Error
+    # Raised if the allow_disk_use option is specified for an operation but the server
+    # selected does not support this option.
+    class UnsupportedAllowDiskUse < Error
```

Comment:
I think I would prefer a single exception class for all of the unsupported options, so that our public API is more flexible with respect to future changes (i.e. in case there is a way of generically validating options in the future).

Reply:
Okay! I'm going to merge this and refactor.